src/rounded_rect.rs

```diff
@@ -149,74 +161,178 @@ impl Shape for RoundedRect {
+                    y: self.rect.y0 + radii.top_left,
+                },
+                Vec2 {
+                    x: radii.top_left,
+                    y: radii.top_left,
                 },
             ),
             build_arc_iter(
                 3,
                 Point {
-                    x: self.rect.x1 - radius,
-                    y: self.rect.y0 + radius,
+                    x: self.rect.x1 - radii.top_right,
+                    y: self.rect.y0 + radii.top_right,
+                },
+                Vec2 {
+                    x: radii.top_right,
+                    y: radii.top_right,
                 },
             ),
             build_arc_iter(
                 0,
                 Point {
-                    x: self.rect.x1 - radius,
-                    y: self.rect.y1 - radius,
+                    x: self.rect.x1 - radii.bottom_right,
+                    y: self.rect.y1 - radii.bottom_right,
+                },
+                Vec2 {
+                    x: radii.bottom_right,
+                    y: radii.bottom_right,
                 },
             ),
             build_arc_iter(
                 1,
                 Point {
-                    x: self.rect.x0 + radius,
-                    y: self.rect.y1 - radius,
+                    x: self.rect.x0 + radii.bottom_left,
+                    y: self.rect.y1 - radii.bottom_left,
+                },
+                Vec2 {
+                    x: radii.bottom_left,
+                    y: radii.bottom_left,
                 },
             ),
         ];
 
         let rect = RectPathIter {
             rect: self.rect,
             ix: 0,
-            radius: self.radius,
+            radii,
         };
 
         RoundedRectPathIter { idx: 0, rect, arcs }
     }
 
     #[inline]
     fn area(&self) -> f64 {
-        let radius = self.radius();
-        self.rect.area() - (4.0 - PI) * radius * radius
+        // A corner is a quarter-circle, i.e.
+        // .............#
+        // .       ######
+        // .    #########
+        // .  ###########
+        // . ############
+        // .#############
+        // ##############
+        // |-----r------|
+        // For each corner, we need to subtract the square that bounds this
+        // quarter-circle, and add back in the area of quarter circle.
+
+        let radii = self.radii();
+
+        // Start with the area of the bounding rect
+        let mut area = self.rect.area();
+
+        // Do for each corner
+        for radius in [
+            radii.top_left,
+            radii.top_right,
+            radii.bottom_right,
+            radii.bottom_left,
+        ]
+        .iter()
+        {
+            // The following would subtract the square and add the quarter-
+            // circle...
+            // area -= radius * radius;
+            // area += FRAC_PI_4 * radius * radius;
+
+            // ...but it can be simplified as:
+            area += (FRAC_PI_4 - 1.0) * radius * radius;
```

Comment:
I believe using `sum` instead of `for` would be slightly more ergonomic, but this is also ok.